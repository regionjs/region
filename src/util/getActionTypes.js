const generator = (name, type) => (name ? `@${name}/${type}` : `@region/${type}`);

export default name => ({
  LOAD_START: generator(name, 'LOAD'),
  SET: generator(name, 'SET'),
});
