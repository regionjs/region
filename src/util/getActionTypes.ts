import { Name } from '../types/types';
type Type = 'LOAD' | 'SET' | 'RESET';

const generator = (name: Name, type: Type) => (name ? `@${name}/${type}` : `@region/${type}`);

export default (name: Name) => ({
  LOAD: generator(name, 'LOAD'),
  SET: generator(name, 'SET'),
  RESET: generator(name, 'RESET'),
});
