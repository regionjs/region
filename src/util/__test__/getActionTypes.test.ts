import getActionTypes from '../getActionTypes';

describe('getActionTypes', () => {
  test('getActionTypes', () => {
    // @ts-ignore
    expect(getActionTypes()).toMatchSnapshot();
    expect(getActionTypes('name')).toMatchSnapshot();
  });
});
