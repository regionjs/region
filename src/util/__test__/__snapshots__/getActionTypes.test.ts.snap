// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getActionTypes getActionTypes 1`] = `
Object {
  "LOAD": "@region/LOAD",
  "RESET": "@region/RESET",
  "SET": "@region/SET",
}
`;

exports[`getActionTypes getActionTypes 2`] = `
Object {
  "LOAD": "@name/LOAD",
  "RESET": "@name/RESET",
  "SET": "@name/SET",
}
`;
